app/controllers/application_controller.rb

```diff
@@ -1,2 +1,10 @@
 class ApplicationController < ActionController::Base
+  protect_from_forgery with: :exception
+
+  before_action :update_allowed_parameters, if: :devise_controller?
+
+  protected
+  def update_allowed_parameters
+    devise_parameter_sanitizer.permit(:sign_up) { |u| u.permit(:firstname, :lastname, :email, :password, :country, :contact)}
```

Comment:
let's not do this in application controller, rather in registrations controller for devise to maintain modularity

Reply:
Devise didn't have a registrations controller added in the controllers, i made one on my own to override any devise's default methods if needed. So should add them in a registrations controller that i made myself?

Reply:
Please go through documentation to see how to override registration controller for devise and do it there